ui/webui/src/components/storage/InstallationDestination.jsx

```diff
@@ -165,35 +173,26 @@ const LocalStandardDisks = ({ idPrefix, setIsFormValid, onAddErrorNotification }
         />
     );
 
-    const diskSelectionLabel = (
-        <Label
-          color="blue"
-          id="installation-destination-table-label"
-        >
-            {cockpit.format(
-                cockpit.ngettext("$0 (of $1) disk selected", "$0 (of $1) disks selected", selectedDisksCnt),
-                selectedDisksCnt,
-                totalDisksCnt
-            )}
-        </Label>
-    );
-
     const rescanDisksButton = (
         <Button
-          aria-label={_("Rescan disks")}
+          aria-label={_("Discover disks")}
           id={idPrefix + "-rescan-disks"}
+          isLoading={isDiscoveringDisks}
+          variant="secondary"
           onClick={() => {
               scanDevicesWithTask().then(res => {
                   runStorageTask({
                       task: res[0],
                       onSuccess: () => resetPartitioning().then(() => setRefreshCnt(refreshCnt + 1), onAddErrorNotification),
                       onFail: onAddErrorNotification
                   });
-              });
+              })
+                      .finally(() => { setIsDiscoveringDisks(false) });
+              setIsDiscoveringDisks(true);
+              setSelectionForAllDisks(false);
```

Comment:
Shoudn't we set the state before we start the discovery?

Reply:
Done.